### PR TITLE
Removing causes of test unstability

### PR DESCRIPTION
**Pull request by odockal**

### CODE REVIEW

File: tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerCodeAssistTest.java [hunk 1]
```diff
@@ -218,20 +220,20 @@ public void after() {
...
-		ProjectExplorer pe = new ProjectExplorer();
-		pe.open();
-		new DefaultTreeItem(prj, "ftl", "empty.ftl").doubleClick();				
-		
-		// editor focus bug workaround 
-		pe.open();
-		new DefaultTreeItem(prj, "ftl", "empty.ftl").doubleClick();				
+		PackageExplorer explorer = new PackageExplorer();
+		Project project = explorer.getProject(projectName);
+		project.expand();
```
Comment (rawagner) [inline tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerCodeAssistTest.java]: you dont need to expand & refresh...just call project.getProjectItem()

File: tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerEditorTest.java [hunk 2]
```diff
@@ -66,11 +68,13 @@ public void freeMarkerTest() {
 
 	private void openFTLFileInEditor() {
 		
-		ProjectExplorer pe = new ProjectExplorer();
-		pe.open();
-
+		PackageExplorer explorer = new PackageExplorer();
+		Project project = explorer.getProject(projectName);
+		project.expand();
```
Comment (rawagner) [inline tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerEditorTest.java]: same situation as mentioned above

File: tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreemarkerDirectiveTest.java [hunk 3]
```diff
@@ -117,11 +115,13 @@ public void ftlDirectiveOutlineTest() {
 	
 	private void openFTLFileInEditor(String file, String... outline) {
 		
-		ProjectExplorer pe = new ProjectExplorer();
-		pe.open();
-
+		PackageExplorer explorer = new PackageExplorer();
+		Project project = explorer.getProject(projectName);
+		project.expand();
```
Comment (rawagner) [inline tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreemarkerDirectiveTest.java]: same situation as mentioned above

File: tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreemarkerTest.java [hunk 4]
```diff
@@ -263,11 +265,7 @@ public void emptyErrorLog() {
 		
 		LogView elv = new LogView();
 		elv.open();
-		try {
-			new ContextMenu("Delete Log").select();
-		} catch (RedDeerException e) {
-			// do nothing
-		}
+		elv.clearLog();
```
Comment (rawagner) [inline tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreemarkerTest.java]: clear log is different action than delete log. Are you sure you want to just clear it ?

File: tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerEditorTest.java [hunk 5]
```diff
@@ -109,7 +113,7 @@ private void checkFreemMarkerOutput() {
 		ProjectExplorer pe = new ProjectExplorer();
 		pe.open();
 
-		new DefaultTreeItem(prj,"src","org.jboss.tools.freemarker.testprj","FMTest.java").select();
+		new DefaultTreeItem(projectName, "src","org.jboss.tools.freemarker.testprj","FMTest.java").select();
```
Comment (rawagner) [inline tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerEditorTest.java]: you should use project.getProjectItem()

File: tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerCodeAssistTest.java (hunk 1 above)
Comment (odockal) [inline tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerCodeAssistTest.java]: Actually I do, it is only workaround for situation when I get treeItem from PackageExplorer, but in background there is still some thread updating the package/project explorer tree and thus the index of saved tree item is not what was requested. (I can just assume that newly loaded nodes are given used indexies and those of existing nodes are increased). 
For example this exception, It occurs time to time:
org.jboss.reddeer.core.exception.CoreLayerException: There are no items matching matcher tree item text on position <0> matches "import-call-directive.ftl"
	at org.jboss.reddeer.core.lookup.TreeItemLookup.getTreeItems(TreeItemLookup.java:132)
	at org.jboss.reddeer.core.lookup.TreeItemLookup.getTreeItems(TreeItemLookup.java:84)
	at org.jboss.reddeer.core.lookup.TreeItemLookup.getTreeItem(TreeItemLookup.java:63)
	at org.jboss.reddeer.swt.impl.tree.DefaultTreeItem.<init>(DefaultTreeItem.java:47)
	at org.jboss.reddeer.swt.impl.tree.DefaultTreeItem.<init>(DefaultTreeItem.java:37)
	at org.jboss.tools.freemarker.ui.bot.test.FreemarkerDirectiveTest.openFTLFileInEditor(FreemarkerDirectiveTest.java:124)
	at org.jboss.tools.freemarker.ui.bot.test.FreemarkerDirectiveTest.importCallDirectiveOutlineTest(FreemarkerDirectiveTest.java:96)

File: tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerEditorTest.java (hunk 5 above)
Comment (odockal) [inline tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerEditorTest.java]: Done.

**Pull request merged**